projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/repository/ProjectRepository.java

```diff
@@ -30,4 +32,6 @@ public interface ProjectRepository {
    * @since 1.0.0
    */
   List<Project> find(ProjectCode projectCode);
+
+  Optional<Project> find(ProjectId projectId);
```

Comment:
because for project codes we do not enforce the singularity in queries, though from an application point we know it should be not more than one.

the project id is the entity id, that is per definition unique.